points[9] right_corners
points[9] left_corners
